### DISCUSSION TITLE
Add option to control video quality

### DISCUSSION THREAD

**Discussion by zhihua-zheng** [ahuang11/streamjoy#24] [Ideas]
Hi,

This is a great package! As of now, I don't see an option to let user set the quality of video. Is it possible to do so?
I think this thread may be helpful: https://github.com/imageio/imageio/issues/1062

**Comment by ahuang11** [comment]
I'm glad you like it!

If I'm not mistaken, you can set `write_kwargs` to modify `write_frame()`
https://github.com/ahuang11/streamjoy/blob/main/streamjoy/streams.py#L765

https://github.com/ahuang11/streamjoy/blob/main/streamjoy/streams.py#L85-L88

Let me know if that helps!

**Reply by zhihua-zheng** [reply]
Tried passing `write_kwarg`, it didn't work. `write_frame()` doesn't support quality-related kwargs.
https://imageio.readthedocs.io/en/stable/_autosummary/imageio.plugins.pyav.PyAVPlugin.write_frame.html

I think we need to set it manually like `buf._video_stream.options = {'crf': '18'}`, according to [this comment](https://github.com/imageio/imageio/issues/1062#issuecomment-1975584390)

**Reply by ahuang11** [reply]
Hmm, I'm hesitant to touch internal variables like that.

Can you check if this works?
```python
with iio.imopen(temp_output, "w", plugin="pyav", options={'crf': '18'}) as writer:
    writer.init_video_stream("libx264", fps=fps)
    ...
    writer.write(batch)
```

If so, I can add `imopen_kwargs`
https://github.com/ahuang11/streamjoy/blob/main/streamjoy/streams.py#L1101

**Reply by zhihua-zheng** [reply]
This doesn't work. As others have reported, I get `Some options were not used: {'crf': '18'}`

**Reply by ahuang11** [reply]
I see. Would you be interested in putting a pull request? 

I think a minor change like this:
```python
if "crf" in write_kwargs:
    buf._video_stream.options = {'crf': str(write_kwargs.pop("crf"))}
```
https://github.com/ahuang11/streamjoy/blob/main/streamjoy/streams.py#L784-L785

https://github.com/ahuang11/streamjoy/blob/main/HOWTOCONTRIBUTE.md

**Reply by zhihua-zheng** [reply]
Yes, I will give it a try later today!

**Reply by ahuang11** [reply]
It's now available in v0.0.8
